Mathlib/Analysis/SpecialFunctions/Elliptic/Weierstrass.lean

```diff
@@ -180,102 +180,105 @@ lemma ℘_bound (r : ℝ) (hr : r > 0) (s : ℂ) (hs : ‖s‖ < r) (l : ℂ) (h
+/-- The Weierstrass `℘` function with the `l₀`-term missing.
 This is mainly a tool for calculations where one would want to omit a diverging term. -/
-def ℘Except (l₀ : ℂ) (z : ℂ) : ℂ :=
+def weierstrassPExcept (l₀ : ℂ) (z : ℂ) : ℂ :=
   ∑' l : L.lattice, if l = l₀ then 0 else (1 / (z - l) ^ 2 - 1 / l ^ 2)
 
-set_option linter.flexible false in -- simp followed by positivity
-lemma hasSumLocallyUniformly_℘Except (l₀ : ℂ) :
+scoped notation3 "℘[" L:max " - " l₀ "]" => weierstrassPExcept L l₀
+
+lemma hasSumLocallyUniformly_weierstrassPExcept (l₀ : ℂ) :
     HasSumLocallyUniformly
       (fun (l : L.lattice) (z : ℂ) ↦ if l.1 = l₀ then 0 else (1 / (z - l) ^ 2 - 1 / l ^ 2))
-      (L.℘Except l₀) := by
+      ℘[L - l₀] := by
   refine L.hasSumLocallyUniformly_aux (u := (10 * · * ‖·‖ ^ (-3 : ℝ))) _
     (fun _ _ ↦ (ZLattice.summable_norm_rpow _ _ (by simp; norm_num)).mul_left _) fun r hr ↦
     Filter.eventually_atTop.mpr ⟨2 * r, ?_⟩
   rintro _ h s hs l rfl
   split_ifs
-  · simp; positivity
-  · exact ℘_bound r hr s hs l h
+  · simpa using show 0 ≤ 10 * r * (‖↑l‖ ^ 3)⁻¹ by positivity
+  · exact weierstrassP_bound r hr s hs l h
 
-lemma hasSum_℘Except (l₀ : ℂ) (z : ℂ) :
+lemma hasSum_weierstrassPExcept (l₀ : ℂ) (z : ℂ) :
     HasSum (fun l : L.lattice ↦ if l = l₀ then 0 else (1 / (z - l) ^ 2 - 1 / l ^ 2))
-      (L.℘Except l₀ z) :=
-  (L.hasSumLocallyUniformly_℘Except l₀).hasSum
+      (℘[L - l₀] z) :=
+  (L.hasSumLocallyUniformly_weierstrassPExcept l₀).hasSum
 
 /- `℘Except l₀` is differentiable on non-lattice points and `l₀`. -/
-lemma differentiableOn_℘Except (l₀ : ℂ) :
-    DifferentiableOn ℂ (L.℘Except l₀) (L.lattice \ {l₀})ᶜ := by
-  refine (L.hasSumLocallyUniformly_℘Except l₀).hasSumLocallyUniformlyOn.differentiableOn
+lemma differentiableOn_weierstrassPExcept (l₀ : ℂ) :
+    DifferentiableOn ℂ ℘[L - l₀] (L.lattice \ {l₀})ᶜ := by
+  refine (L.hasSumLocallyUniformly_weierstrassPExcept l₀).hasSumLocallyUniformlyOn.differentiableOn
     (.of_forall fun s ↦ .fun_sum fun i hi ↦ ?_)
     (L.isClosed_of_subset_lattice Set.diff_subset).isOpen_compl
   split_ifs
   · simp
   · exact .sub (.div (by fun_prop) (by fun_prop) (by aesop (add simp sub_eq_zero))) (by fun_prop)
 
-lemma ℘Except_neg (l₀ : ℂ) (z : ℂ) : L.℘Except l₀ (-z) = L.℘Except (-l₀) z := by
-  simp only [℘Except]
+lemma weierstrassPExcept_neg (l₀ : ℂ) (z : ℂ) :
+    ℘[L - l₀] (-z) = ℘[L - -l₀] z := by
+  simp only [weierstrassPExcept]
   rw [← (Equiv.neg L.lattice).tsum_eq]
   congr! 3 with l
   · simp [neg_eq_iff_eq_neg]
   simp
   ring
 
-end ℘Except
+end weierstrassPExcept
+
+section weierstrassP
 
-section ℘
+/-- The Weierstrass `℘` function. This has the notation `℘[L]` in `namespace PeriodPairs`. -/
+def weierstrassP (z : ℂ) : ℂ := ∑' l : L.lattice, (1 / (z - l) ^ 2 - 1 / l ^ 2)
 
-/-- The Weierstrass ℘ function. -/
-def ℘ (z : ℂ) : ℂ := ∑' l : L.lattice, (1 / (z - l) ^ 2 - 1 / l ^ 2)
+scoped notation3 "℘[" L "]" => weierstrassP L
```

Comment:
this notation looks like it might clash with the inbuilt getElem notation...

Reply:
I don't think this is a big concern because it is scoped in `PeriodPairs`, and if anyone opened `PeriodPairs` while having a variable named `℘` it is probably their fault?